frontend/generateCard.js

```diff
@@ -1,65 +1,74 @@
 const cardHolder = document.getElementById('cardHolder');
 const cards = cardHolder.getElementsByClassName('card');
+let cardsData = {}
+let keywordData = [];
+let descriptionData = [];
 
-// Array of card titles
-const titles = [
-    'Deathtouch',
-    'Haste',
-    'Flying',
-    'Reach',
-    'Hexproof',
-    'Double Strike',
-    'First Strike',
-    'Lifelink'
-];
+// Fetch card data from API
+async function fetchCard() {
+    const url = 'http://127.0.0.1:5000/keyword-abilities';
 
-// Array of descriptions corresponding to the titles
-const descriptions = [
-    'Deathtouch (Any amount of damage this deals to a creature is enough to destroy it.)',
-    'Haste (This creature can attack and {T} as soon as it comes under your control.)',
-    "Flying (This creature can't be blocked except by creatures with flying and/or reach.)",
-    "Reach (This creature can block creatures with flying.)",
-    "Hexproof (This permanent can't be the target of spells or abilities your opponents control.)",
-    "Double strike (This creature deals both first-strike and regular combat damage.)",
-    "First strike (This creature deals combat damage before creatures without first strike.)",
-    "Lifelink (Damage dealt by this creature also causes you to gain that much life.)"
-];
+    try {
+        const response = await fetch(url);
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        const json = await response.json();
+        cardsData = json; //Full api data
 
-// Function to create a card element with a title and description
-function makeCard(title, description) {
-    // Create a new div element for the card
+        // Taking the keyword titles and descriptions
+        cardsData.forEach(card => {
```

Comment:
I don't think we need to extract the attributes from the parsed json object here. We should remove the arrays and move the cardsData variable out from the method and access the attributes using a for each loop when constructing the cards. Then we can remove some of the logic and reduce the complexity and optimise it a bit as no data need to be copied.

Reply:
Now the fetchCardData fetches data directly without putting it into a variable, returning the extracted .json object directly.
The cardData variable gets declared and defined with the fetch data in the renderCards() function, then creates and renders a card forEach object in cardData